src/internal/connector/onedrive/collection.go

```diff
@@ -251,7 +252,32 @@ func (oc *Collection) populateItems(ctx context.Context) {
 
 				for i := 1; i <= maxRetries; i++ {
 					_, itemData, err = oc.itemReader(oc.itemClient, item)
-					if err == nil || graph.IsErrTimeout(err) == nil {
+					if err == nil {
+						break
+					}
+
+					if graph.IsErrUnauthorized(err) != nil {
+						// assume unauthorized requests are a sign of an expired
+						// jwt token, and that we've overrun the available window
+						// to download the actual file.  Re-downloading the item
+						// will refresh that download url.
+						di, diErr := getDriveItem(ctx, oc.service, oc.driveID, itemID)
+						if diErr != nil {
+							err = errors.Wrap(diErr, "retrieving expired item")
+							break
+						}
+
+						item = di
+
+						continue
+
+					} else if graph.IsErrTimeout(err) == nil &&
+						graph.IsErrThrottled(err) == nil &&
+						graph.IsSericeUnavailable(err) == nil {
+						// TODO: graphAPI will provides headers that state the duration to wait
+						// in order to succeed again.  The one second sleep won't cut it here.
```

Comment:
I believe the `Retry-After` header is only available for throttled requests, but they are handled internally within the [default retry handler](https://github.com/microsoft/kiota-http-go/blob/af8d9d1d0fbf3841ff2dc1d87b754ef0d47b2368/retry_handler.go#L127) within kiota.

Reply:
That gets handled in _some_ cases.  Fun fact about the default retry handler: it doesn't work on GET requests.

```
func (middleware RetryHandler) isRetriableRequest(req *nethttp.Request) bool {
	isBodiedMethod := req.Method == "POST" || req.Method == "PUT" || req.Method == "PATCH"
```

Reply:
> Nevermind me! I should read the whole func before I respond. Thanks for the reminder.

Same :). I made the same assumption when I was reading the code with @meain last time